Bridgestan failure with Hessian AD enabled

I'm not certain if this is an issue with bridgestan or coming from somewhere else in the toolchain, but if I install gxx from conda-forge, install bridgestan from pypi, and then compile a model requesting `BRIDGESTAN_AD_HESSIAN=true`, then it errors if I have an expression like `target += y` for a vector `y`. It doesn't fail if I use `target += sum(y)`, and it also doesn't fail if I don't set `BRIDGESTAN_AD_HESSIAN=true`.

Here's a reproducible example:
`test.stan`:
```stan
data {
  int<lower=1> N;
}
parameters {
  vector[N] y;
}
model {
  target += y;    
}
```

Script:
```bash
conda create -n bridgestan-test -c conda-forge python=3.11 gxx=13.2.0 pip
conda activate bridgestan-test
python -m pip install bridgestan==2.5.0
python -c '
import bridgestan as bs
stan_file = "./test.stan"
bs.compile_model(stan_file, make_args=["BRIDGESTAN_AD_HESSIAN=true"])
'
```

Error:
```
Traceback (most recent call last):
  File "<string>", line 4, in <module>
  File "/home/sethaxen/software/mambaforge/envs/bridgestan-test/lib/python3.11/site-packages/bridgestan/compile.py", line 142, in compile_model
    raise RuntimeError(error)
RuntimeError: Command make BRIDGESTAN_AD_HESSIAN=true STANCFLAGS=--include-paths=. /home/sethaxen/bridgestan_test/test_model.so failed with code 2.
stdout:

--- Translating Stan model to C++ code ---
./bin/stanc --include-paths=. --o=/home/sethaxen/bridgestan_test/test.hpp /home/sethaxen/bridgestan_test/test.stan

--- Compiling C++ code ---
g++ -std=c++17 -pthread -D_REENTRANT -Wno-sign-compare -Wno-ignored-attributes -Wno-class-memaccess      -I ./stan/lib/stan_math/lib/tbb_2020.3/include    -O3 -I ./stan/src -I ./stan/lib/rapidjson_1.1.0/ -I ./stan/lib/stan_math/ -I ./stan/lib/stan_math/lib/eigen_3.4.0 -I ./stan/lib/stan_math/lib/boost_1.84.0 -I ./stan/lib/stan_math/lib/sundials_6.1.1/include -I ./stan/lib/stan_math/lib/sundials_6.1.1/src/sundials -fPIC -fvisibility=hidden -fvisibility-inlines-hidden    -DBOOST_DISABLE_ASSERTS         -DBRIDGESTAN_EXPORT -DSTAN_MODEL_FVAR_VAR -DBRIDGESTAN_AD_HESSIAN  -c  -x c++ -o /home/sethaxen/bridgestan_test/test.o /home/sethaxen/bridgestan_test/test.hpp
rm /home/sethaxen/bridgestan_test/test.hpp

stderr:
In file included from ./stan/lib/stan_math/lib/boost_1.84.0/boost/numeric/ublas/traits.hpp:21,
                 from ./stan/lib/stan_math/lib/boost_1.84.0/boost/numeric/ublas/storage.hpp:27,
                 from ./stan/lib/stan_math/lib/boost_1.84.0/boost/numeric/ublas/vector.hpp:21,
                 from ./stan/lib/stan_math/lib/boost_1.84.0/boost/numeric/odeint/util/ublas_wrapper.hpp:23,
                 from ./stan/lib/stan_math/lib/boost_1.84.0/boost/numeric/odeint.hpp:25,
                 from ./stan/lib/stan_math/stan/math/prim/functor/ode_rk45.hpp:9,
                 from ./stan/lib/stan_math/stan/math/prim/functor/integrate_ode_rk45.hpp:6,
                 from ./stan/lib/stan_math/stan/math/prim/functor.hpp:16,
                 from ./stan/lib/stan_math/stan/math/rev/fun.hpp:204,
                 from ./stan/lib/stan_math/stan/math/rev.hpp:12,
                 from ./stan/lib/stan_math/stan/math.hpp:19,
                 from ./stan/src/stan/model/model_header.hpp:4,
                 from /home/sethaxen/bridgestan_test/test.hpp:2:
./stan/lib/stan_math/lib/boost_1.84.0/boost/numeric/ublas/detail/iterator.hpp:111:21: warning: 'template<class _Category, class _Tp, class _Distance, class _Pointer, class _Reference> struct std::iterator' is deprecated [-Wdeprecated-declarations]
  111 |         public std::iterator<IC, T> {
      |                     ^~~~~~~~
In file included from /home/sethaxen/software/mambaforge/envs/bridgestan-test/lib/gcc/x86_64-conda-linux-gnu/13.2.0/include/c++/bits/stl_algobase.h:65,
                 from /home/sethaxen/software/mambaforge/envs/bridgestan-test/lib/gcc/x86_64-conda-linux-gnu/13.2.0/include/c++/bits/specfun.h:43,
                 from /home/sethaxen/software/mambaforge/envs/bridgestan-test/lib/gcc/x86_64-conda-linux-gnu/13.2.0/include/c++/cmath:3699,
                 from ./stan/lib/stan_math/lib/eigen_3.4.0/Eigen/src/Core/util/Macros.h:679,
                 from ./stan/lib/stan_math/lib/eigen_3.4.0/Eigen/Core:19,
                 from ./stan/lib/stan_math/lib/eigen_3.4.0/Eigen/Dense:1,
                 from ./stan/lib/stan_math/stan/math/prim/fun/Eigen.hpp:22,
                 from ./stan/lib/stan_math/stan/math/rev.hpp:4:
/home/sethaxen/software/mambaforge/envs/bridgestan-test/lib/gcc/x86_64-conda-linux-gnu/13.2.0/include/c++/bits/stl_iterator_base_types.h:127:34: note: declared here
  127 |     struct _GLIBCXX17_DEPRECATED iterator
      |                                  ^~~~~~~~
./stan/lib/stan_math/lib/boost_1.84.0/boost/numeric/ublas/detail/iterator.hpp:149:21: warning: 'template<class _Category, class _Tp, class _Distance, class _Pointer, class _Reference> struct std::iterator' is deprecated [-Wdeprecated-declarations]
  149 |         public std::iterator<IC, T> {
      |                     ^~~~~~~~
/home/sethaxen/software/mambaforge/envs/bridgestan-test/lib/gcc/x86_64-conda-linux-gnu/13.2.0/include/c++/bits/stl_iterator_base_types.h:127:34: note: declared here
  127 |     struct _GLIBCXX17_DEPRECATED iterator
      |                                  ^~~~~~~~
./stan/lib/stan_math/lib/boost_1.84.0/boost/numeric/ublas/detail/iterator.hpp:204:21: warning: 'template<class _Category, class _Tp, class _Distance, class _Pointer, class _Reference> struct std::iterator' is deprecated [-Wdeprecated-declarations]
  204 |         public std::iterator<IC, T> {
      |                     ^~~~~~~~
/home/sethaxen/software/mambaforge/envs/bridgestan-test/lib/gcc/x86_64-conda-linux-gnu/13.2.0/include/c++/bits/stl_iterator_base_types.h:127:34: note: declared here
  127 |     struct _GLIBCXX17_DEPRECATED iterator
      |                                  ^~~~~~~~
In file included from ./stan/lib/stan_math/stan/math/rev/fun/accumulator.hpp:7,
                 from ./stan/lib/stan_math/stan/math/rev/fun.hpp:10:
./stan/lib/stan_math/stan/math/prim/fun/accumulator.hpp: In instantiation of 'void stan::math::accumulator<T, <template-parameter-1-2> >::add(const S&) [with S = Eigen::Matrix<stan::math::fvar<stan::math::var_value<double> >, -1, 1>; stan::require_matrix_t<S>* <anonymous> = 0; T = stan::math::fvar<stan::math::var_value<double> >; <template-parameter-1-2> = void]':
/home/sethaxen/bridgestan_test/test.hpp:88:0:   required from 'stan::scalar_type_t<T2> test_model_namespace::test_model::log_prob_impl(VecR&, VecI&, std::ostream*) const [with bool propto__ = false; bool jacobian__ = false; VecR = Eigen::Matrix<stan::math::fvar<stan::math::var_value<double> >, -1, 1>; VecI = Eigen::Matrix<int, -1, 1>; stan::require_vector_like_t<VecR>* <anonymous> = 0; stan::require_vector_like_vt<std::is_integral, VecI>* <anonymous> = 0; stan::require_not_st_var<VecR>* <anonymous> = 0; stan::scalar_type_t<T2> = stan::math::fvar<stan::math::var_value<double> >; std::ostream = std::basic_ostream<char>]'
/home/sethaxen/bridgestan_test/test.hpp:332:0:   required from 'T_ test_model_namespace::test_model::log_prob(Eigen::Matrix<T_a, -1, 1>&, std::ostream*) const [with bool propto__ = false; bool jacobian__ = false; T_ = stan::math::fvar<stan::math::var_value<double> >; std::ostream = std::basic_ostream<char>]'
./stan/src/stan/model/model_base_crtp.hpp:241:0:   required from 'stan::math::fvar<stan::math::var_value<double> > stan::model::model_base_crtp<M>::log_prob(Eigen::Matrix<stan::math::fvar<stan::math::var_value<double> >, -1, 1>&, std::ostream*) const [with M = test_model_namespace::test_model; std::ostream = std::basic_ostream<char>]'
./stan/src/stan/model/model_base_crtp.hpp:238:0:   required from here
./stan/lib/stan_math/stan/math/prim/fun/accumulator.hpp:53:35: error: no matching function for call to 'sum(const Eigen::Matrix<stan::math::fvar<stan::math::var_value<double> >, -1, 1>&)'
   53 |     buf_.push_back(stan::math::sum(m));
      |                    ~~~~~~~~~~~~~~~^~~
In file included from ./stan/lib/stan_math/stan/math/prim/meta/possibly_sum.hpp:4,
                 from ./stan/lib/stan_math/stan/math/prim/meta.hpp:119,
                 from ./stan/lib/stan_math/stan/math/rev/core/accumulate_adjoints.hpp:4,
                 from ./stan/lib/stan_math/stan/math/rev/core.hpp:4,
                 from ./stan/lib/stan_math/stan/math/rev.hpp:10:
./stan/lib/stan_math/stan/math/prim/fun/sum.hpp:21:10: note: candidate: 'template<class T, stan::require_stan_scalar_t<T>* <anonymous> > T stan::math::sum(T&&)'
   21 | inline T sum(T&& m) {
      |          ^~~
./stan/lib/stan_math/stan/math/prim/fun/sum.hpp:21:10: note:   template argument deduction/substitution failed:
In file included from /home/sethaxen/software/mambaforge/envs/bridgestan-test/lib/gcc/x86_64-conda-linux-gnu/13.2.0/include/c++/bits/stl_pair.h:60,
                 from /home/sethaxen/software/mambaforge/envs/bridgestan-test/lib/gcc/x86_64-conda-linux-gnu/13.2.0/include/c++/bits/stl_algobase.h:64:
/home/sethaxen/software/mambaforge/envs/bridgestan-test/lib/gcc/x86_64-conda-linux-gnu/13.2.0/include/c++/type_traits: In substitution of 'template<bool _Cond, class _Tp> using std::enable_if_t = typename std::enable_if::type [with bool _Cond = false; _Tp = void]':
./stan/lib/stan_math/stan/math/prim/meta/require_helpers.hpp:19:7:   required by substitution of 'template<class Check> using stan::require_t = std::enable_if_t<Check::value> [with Check = stan::is_stan_scalar<Eigen::Matrix<stan::math::fvar<stan::math::var_value<double> >, -1, 1> >]'
./stan/lib/stan_math/stan/math/prim/meta/is_stan_scalar.hpp:39:7:   required by substitution of 'template<class T> using stan::require_stan_scalar_t = stan::require_t<stan::is_stan_scalar<typename std::decay<_Tp>::type> > [with T = const Eigen::Matrix<stan::math::fvar<stan::math::var_value<double> >, -1, 1>&]'
./stan/lib/stan_math/stan/math/prim/fun/sum.hpp:20:51:   required from 'void stan::math::accumulator<T, <template-parameter-1-2> >::add(const S&) [with S = Eigen::Matrix<stan::math::fvar<stan::math::var_value<double> >, -1, 1>; stan::require_matrix_t<S>* <anonymous> = 0; T = stan::math::fvar<stan::math::var_value<double> >; <template-parameter-1-2> = void]'
/home/sethaxen/bridgestan_test/test.hpp:88:0:   required from 'stan::scalar_type_t<T2> test_model_namespace::test_model::log_prob_impl(VecR&, VecI&, std::ostream*) const [with bool propto__ = false; bool jacobian__ = false; VecR = Eigen::Matrix<stan::math::fvar<stan::math::var_value<double> >, -1, 1>; VecI = Eigen::Matrix<int, -1, 1>; stan::require_vector_like_t<VecR>* <anonymous> = 0; stan::require_vector_like_vt<std::is_integral, VecI>* <anonymous> = 0; stan::require_not_st_var<VecR>* <anonymous> = 0; stan::scalar_type_t<T2> = stan::math::fvar<stan::math::var_value<double> >; std::ostream = std::basic_ostream<char>]'
/home/sethaxen/bridgestan_test/test.hpp:332:0:   required from 'T_ test_model_namespace::test_model::log_prob(Eigen::Matrix<T_a, -1, 1>&, std::ostream*) const [with bool propto__ = false; bool jacobian__ = false; T_ = stan::math::fvar<stan::math::var_value<double> >; std::ostream = std::basic_ostream<char>]'
./stan/src/stan/model/model_base_crtp.hpp:241:0:   required from 'stan::math::fvar<stan::math::var_value<double> > stan::model::model_base_crtp<M>::log_prob(Eigen::Matrix<stan::math::fvar<stan::math::var_value<double> >, -1, 1>&, std::ostream*) const [with M = test_model_namespace::test_model; std::ostream = std::basic_ostream<char>]'
./stan/src/stan/model/model_base_crtp.hpp:238:0:   required from here
/home/sethaxen/software/mambaforge/envs/bridgestan-test/lib/gcc/x86_64-conda-linux-gnu/13.2.0/include/c++/type_traits:2610:11: error: no type named 'type' in 'struct std::enable_if<false, void>'
 2610 |     using enable_if_t = typename enable_if<_Cond, _Tp>::type;
      |           ^~~~~~~~~~~
./stan/lib/stan_math/stan/math/prim/fun/accumulator.hpp: In instantiation of 'void stan::math::accumulator<T, <template-parameter-1-2> >::add(const S&) [with S = Eigen::Matrix<stan::math::fvar<stan::math::var_value<double> >, -1, 1>; stan::require_matrix_t<S>* <anonymous> = 0; T = stan::math::fvar<stan::math::var_value<double> >; <template-parameter-1-2> = void]':
/home/sethaxen/bridgestan_test/test.hpp:88:0:   required from 'stan::scalar_type_t<T2> test_model_namespace::test_model::log_prob_impl(VecR&, VecI&, std::ostream*) const [with bool propto__ = false; bool jacobian__ = false; VecR = Eigen::Matrix<stan::math::fvar<stan::math::var_value<double> >, -1, 1>; VecI = Eigen::Matrix<int, -1, 1>; stan::require_vector_like_t<VecR>* <anonymous> = 0; stan::require_vector_like_vt<std::is_integral, VecI>* <anonymous> = 0; stan::require_not_st_var<VecR>* <anonymous> = 0; stan::scalar_type_t<T2> = stan::math::fvar<stan::math::var_value<double> >; std::ostream = std::basic_ostream<char>]'
/home/sethaxen/bridgestan_test/test.hpp:332:0:   required from 'T_ test_model_namespace::test_model::log_prob(Eigen::Matrix<T_a, -1, 1>&, std::ostream*) const [with bool propto__ = false; bool jacobian__ = false; T_ = stan::math::fvar<stan::math::var_value<double> >; std::ostream = std::basic_ostream<char>]'
./stan/src/stan/model/model_base_crtp.hpp:241:0:   required from 'stan::math::fvar<stan::math::var_value<double> > stan::model::model_base_crtp<M>::log_prob(Eigen::Matrix<stan::math::fvar<stan::math::var_value<double> >, -1, 1>&, std::ostream*) const [with M = test_model_namespace::test_model; std::ostream = std::basic_ostream<char>]'
./stan/src/stan/model/model_base_crtp.hpp:238:0:   required from here
./stan/lib/stan_math/stan/math/prim/fun/sum.hpp:33:10: note: candidate: 'template<class T, stan::require_not_var_t<T>* <anonymous> > T stan::math::sum(const std::vector<T1>&)'
   33 | inline T sum(const std::vector<T>& m) {
      |          ^~~
./stan/lib/stan_math/stan/math/prim/fun/sum.hpp:33:10: note:   template argument deduction/substitution failed:
./stan/lib/stan_math/stan/math/prim/fun/accumulator.hpp:53:35: note:   'const Eigen::Matrix<stan::math::fvar<stan::math::var_value<double> >, -1, 1>' is not derived from 'const std::vector<T1>'
   53 |     buf_.push_back(stan::math::sum(m));
      |                    ~~~~~~~~~~~~~~~^~~
./stan/lib/stan_math/stan/math/prim/fun/sum.hpp:46:24: note: candidate: 'template<class T, stan::require_eigen_vt<std::is_arithmetic, T>* <anonymous> > stan::value_type_t<T> stan::math::sum(const T&)'
   46 | inline value_type_t<T> sum(const T& m) {
      |                        ^~~
./stan/lib/stan_math/stan/math/prim/fun/sum.hpp:46:24: note:   template argument deduction/substitution failed:
./stan/lib/stan_math/stan/math/prim/fun/sum.hpp:59:24: note: candidate: 'template<class T, stan::require_eigen_vt<stan::is_complex, T>* <anonymous> > stan::value_type_t<T> stan::math::sum(const T&)'
   59 | inline value_type_t<T> sum(const T& m) {
      |                        ^~~
./stan/lib/stan_math/stan/math/prim/fun/sum.hpp:59:24: note:   template argument deduction/substitution failed:
In file included from ./stan/lib/stan_math/stan/math/rev/fun/accumulator.hpp:6:
./stan/lib/stan_math/stan/math/rev/fun/sum.hpp:26:12: note: candidate: 'template<class Alloc> stan::math::var stan::math::sum(const std::vector<var_value<double>, Alloc>&)'
   26 | inline var sum(const std::vector<var, Alloc>& m) {
      |            ^~~
./stan/lib/stan_math/stan/math/rev/fun/sum.hpp:26:12: note:   template argument deduction/substitution failed:
./stan/lib/stan_math/stan/math/prim/fun/accumulator.hpp:53:35: note:   'const Eigen::Matrix<stan::math::fvar<stan::math::var_value<double> >, -1, 1>' is not derived from 'const std::vector<stan::math::var_value<double>, Alloc>'
   53 |     buf_.push_back(stan::math::sum(m));
      |                    ~~~~~~~~~~~~~~~^~~
./stan/lib/stan_math/stan/math/rev/fun/sum.hpp:47:12: note: candidate: 'template<class T, stan::require_rev_matrix_t<T>* <anonymous> > stan::math::var stan::math::sum(const T&)'
   47 | inline var sum(const T& x) {
      |            ^~~
./stan/lib/stan_math/stan/math/rev/fun/sum.hpp:47:12: note:   template argument deduction/substitution failed:
make: *** [Makefile:78: /home/sethaxen/bridgestan_test/test.o] Error 1
```

Hi @WardBrian, I hit a perhaps unrelated issue that only shows up when compiling for Hessian AD if the target is incremented by a constant in the `model` block.:

`test.stan`
```stan
data {
  int<lower=1> N;
}
parameters {
  vector[N] y;
}
model {
  target += 0;
}
```
Running the same command as in https://github.com/roualdes/bridgestan/issues/237#issue-2370499296 we get the error:
```
Traceback (most recent call last):
  File "<string>", line 4, in <module>
  File "/home/sethaxen/software/mambaforge/envs/bridgestan-test/lib/python3.11/site-packages/bridgestan/compile.py", line 142, in compile_model
    raise RuntimeError(error)
RuntimeError: Command make BRIDGESTAN_AD_HESSIAN=true STANCFLAGS=--include-paths=. /home/sethaxen/bridgestan_test/test_model.so failed with code 2.
stdout:

--- Translating Stan model to C++ code ---
./bin/stanc --include-paths=. --o=/home/sethaxen/bridgestan_test/test.hpp /home/sethaxen/bridgestan_test/test.stan

stderr:
./bin/stanc: line 1: Not: command not found
make: *** [Makefile:73: /home/sethaxen/bridgestan_test/test.hpp] Error 127
```

If it's incremented by a non-constant in an `_lp` function we still get this error, e.g.
`test.stan`:
```stan
functions {
  vector foo_lp(vector y) {
    target += sum(y);
    return y;
  }
}
data {
  int<lower=1> N;
}
parameters {
  vector[N] y;
}
transformed parameters {
  vector[N] x = foo_lp(y);
}
model {
  target += 1;
}
```
This probably also is not a bridgestan issue, but I'm not sure where to report it.

--- a/src/stan/model/model_base.hpp
+++ b/src/stan/model/model_base.hpp
@@ -78,8 +78,7 @@ class model_base : public prob_grad {
    */
   virtual void get_param_names(std::vector<std::string>& names,
                                bool include_tparams = true,
-                               bool include_gqs = true) const
-      = 0;
+                               bool include_gqs = true) const = 0;
   /**
    * WARNING: This function bakes in the assumption that the
    * parameter values are rectangular. This is already not true
@@ -108,8 +107,7 @@ class model_base : public prob_grad {
    */
   virtual void get_dims(std::vector<std::vector<size_t> >& dimss,
                         bool include_tparams = true,
-                        bool include_gqs = true) const
-      = 0;
+                        bool include_gqs = true) const = 0;
   /**
    *  Set the specified sequence to the indexed, scalar, constrained
    *  parameter names.  Each variable is output with a
@@ -150,8 +148,7 @@ class model_base : public prob_grad {
    */
   virtual void constrained_param_names(std::vector<std::string>& param_names,
                                        bool include_tparams = true,
-                                       bool include_gqs = true) const
-      = 0;
+                                       bool include_gqs = true) const = 0;
 
   /**
    * Set the specified sequence of parameter names to the
@@ -180,8 +177,7 @@ class model_base : public prob_grad {
    */
   virtual void unconstrained_param_names(std::vector<std::string>& param_names,
                                          bool include_tparams = true,
-                                         bool include_gqs = true) const
-      = 0;
+                                         bool include_gqs = true) const = 0;
 
   /**
    * Return the log density for the specified unconstrained
@@ -192,8 +188,8 @@ class model_base : public prob_grad {
    * @param[in,out] msgs message stream
    * @return log density for specified parameters
    */
-  virtual double log_prob(Eigen::VectorXd& params_r, std::ostream* msgs) const
-      = 0;
+  virtual double log_prob(Eigen::VectorXd& params_r,
+                          std::ostream* msgs) const = 0;
 
   /**
    * Return the log density for the specified unconstrained
@@ -205,8 +201,7 @@ class model_base : public prob_grad {
    * @return log density for specified parameters
    */
   virtual math::var log_prob(Eigen::Matrix<math::var, -1, 1>& params_r,
-                             std::ostream* msgs) const
-      = 0;
+                             std::ostream* msgs) const = 0;
 
   /**
    * Return the log density for the specified unconstrained
@@ -222,8 +217,7 @@ class model_base : public prob_grad {
    * @return log density for specified parameters
    */
   virtual double log_prob_jacobian(Eigen::VectorXd& params_r,
-                                   std::ostream* msgs) const
-      = 0;
+                                   std::ostream* msgs) const = 0;
 
   /**
    * Return the log density for the specified unconstrained
@@ -239,8 +233,7 @@ class model_base : public prob_grad {
    * @return log density for specified parameters
    */
   virtual math::var log_prob_jacobian(Eigen::Matrix<math::var, -1, 1>& params_r,
-                                      std::ostream* msgs) const
-      = 0;
+                                      std::ostream* msgs) const = 0;
 
   /**
    * Return the log density for the specified unconstrained
@@ -257,8 +250,7 @@ class model_base : public prob_grad {
    * @return log density for specified parameters
    */
   virtual double log_prob_propto(Eigen::VectorXd& params_r,
-                                 std::ostream* msgs) const
-      = 0;
+                                 std::ostream* msgs) const = 0;
 
   /**
    * Return the log density for the specified unconstrained
@@ -270,8 +262,7 @@ class model_base : public prob_grad {
    * @return log density for specified parameters
    */
   virtual math::var log_prob_propto(Eigen::Matrix<math::var, -1, 1>& params_r,
-                                    std::ostream* msgs) const
-      = 0;
+                                    std::ostream* msgs) const = 0;
 
   /**
    * Return the log density for the specified unconstrained
@@ -292,8 +283,7 @@ class model_base : public prob_grad {
    * @return log density for specified parameters
    */
   virtual double log_prob_propto_jacobian(Eigen::VectorXd& params_r,
-                                          std::ostream* msgs) const
-      = 0;
+                                          std::ostream* msgs) const = 0;
 
   /**
    * Return the log density for the specified unconstrained
@@ -309,8 +299,7 @@ class model_base : public prob_grad {
    * @return log density for specified parameters
    */
   virtual math::var log_prob_propto_jacobian(
-      Eigen::Matrix<math::var, -1, 1>& params_r, std::ostream* msgs) const
-      = 0;
+      Eigen::Matrix<math::var, -1, 1>& params_r, std::ostream* msgs) const = 0;
 
   /**
    * Convenience template function returning the log density for the
@@ -358,8 +347,7 @@ class model_base : public prob_grad {
    */
   virtual void transform_inits(const io::var_context& context,
                                Eigen::VectorXd& params_r,
-                               std::ostream* msgs) const
-      = 0;
+                               std::ostream* msgs) const = 0;
 
   /**
    * Convert the specified sequence of unconstrained parameters to a
@@ -382,8 +370,7 @@ class model_base : public prob_grad {
   virtual void write_array(stan::rng_t& base_rng, Eigen::VectorXd& params_r,
                            Eigen::VectorXd& params_constrained_r,
                            bool include_tparams = true, bool include_gqs = true,
-                           std::ostream* msgs = 0) const
-      = 0;
+                           std::ostream* msgs = 0) const = 0;
 
   /**
    * Convert the specified sequence of constrained parameters to a
@@ -398,8 +385,7 @@ class model_base : public prob_grad {
    */
   virtual void unconstrain_array(const Eigen::VectorXd& params_r_constrained,
                                  Eigen::VectorXd& params_r,
-                                 std::ostream* msgs = nullptr) const
-      = 0;
+                                 std::ostream* msgs = nullptr) const = 0;
 
   // TODO(carpenter): cut redundant std::vector versions from here ===
 
@@ -416,8 +402,8 @@ class model_base : public prob_grad {
    * @return log density for specified parameters
    */
   virtual double log_prob(std::vector<double>& params_r,
-                          std::vector<int>& params_i, std::ostream* msgs) const
-      = 0;
+                          std::vector<int>& params_i,
+                          std::ostream* msgs) const = 0;
 
   /**
    * Return the log density for the specified unconstrained
@@ -433,8 +419,7 @@ class model_base : public prob_grad {
    */
   virtual math::var log_prob(std::vector<math::var>& params_r,
                              std::vector<int>& params_i,
-                             std::ostream* msgs) const
-      = 0;
+                             std::ostream* msgs) const = 0;
 
   /**
    * Return the log density for the specified unconstrained
@@ -454,8 +439,7 @@ class model_base : public prob_grad {
    */
   virtual double log_prob_jacobian(std::vector<double>& params_r,
                                    std::vector<int>& params_i,
-                                   std::ostream* msgs) const
-      = 0;
+                                   std::ostream* msgs) const = 0;
 
   /**
    * Return the log density for the specified unconstrained
@@ -475,8 +459,7 @@ class model_base : public prob_grad {
    */
   virtual math::var log_prob_jacobian(std::vector<math::var>& params_r,
                                       std::vector<int>& params_i,
-                                      std::ostream* msgs) const
-      = 0;
+                                      std::ostream* msgs) const = 0;
 
   /**
    * Return the log density for the specified unconstrained
@@ -497,8 +480,7 @@ class model_base : public prob_grad {
    */
   virtual double log_prob_propto(std::vector<double>& params_r,
                                  std::vector<int>& params_i,
-                                 std::ostream* msgs) const
-      = 0;
+                                 std::ostream* msgs) const = 0;
 
   /**
    * Return the log density for the specified unconstrained
@@ -514,8 +496,7 @@ class model_base : public prob_grad {
    */
   virtual math::var log_prob_propto(std::vector<math::var>& params_r,
                                     std::vector<int>& params_i,
-                                    std::ostream* msgs) const
-      = 0;
+                                    std::ostream* msgs) const = 0;
 
   /**
    * Return the log density for the specified unconstrained
@@ -540,8 +521,7 @@ class model_base : public prob_grad {
    */
   virtual double log_prob_propto_jacobian(std::vector<double>& params_r,
                                           std::vector<int>& params_i,
-                                          std::ostream* msgs) const
-      = 0;
+                                          std::ostream* msgs) const = 0;
 
   /**
    * Return the log density for the specified unconstrained
@@ -561,8 +541,7 @@ class model_base : public prob_grad {
    */
   virtual math::var log_prob_propto_jacobian(std::vector<math::var>& params_r,
                                              std::vector<int>& params_i,
-                                             std::ostream* msgs) const
-      = 0;
+                                             std::ostream* msgs) const = 0;
 
   /**
    * Convenience template function returning the log density for the
@@ -617,8 +596,7 @@ class model_base : public prob_grad {
   virtual void transform_inits(const io::var_context& context,
                                std::vector<int>& params_i,
                                std::vector<double>& params_r,
-                               std::ostream* msgs) const
-      = 0;
+                               std::ostream* msgs) const = 0;
 
   /**
    * Convert the specified sequence of unconstrained parameters to a
@@ -643,8 +621,7 @@ class model_base : public prob_grad {
                            std::vector<int>& params_i,
                            std::vector<double>& params_r_constrained,
                            bool include_tparams = true, bool include_gqs = true,
-                           std::ostream* msgs = 0) const
-      = 0;
+                           std::ostream* msgs = 0) const = 0;
 
   /**
    * Convert the specified sequence of constrained parameters to a
@@ -659,8 +636,7 @@ class model_base : public prob_grad {
    */
   virtual void unconstrain_array(
       const std::vector<double>& params_r_constrained,
-      std::vector<double>& params_r, std::ostream* msgs = nullptr) const
-      = 0;
+      std::vector<double>& params_r, std::ostream* msgs = nullptr) const = 0;
 
 #ifdef STAN_MODEL_FVAR_VAR
 
@@ -675,8 +651,7 @@ class model_base : public prob_grad {
    */
   virtual math::fvar<math::var> log_prob(
       Eigen::Matrix<math::fvar<math::var>, -1, 1>& params_r,
-      std::ostream* msgs) const
-      = 0;
+      std::ostream* msgs) const = 0;
 
   /**
    * Return the log density for the specified unconstrained
@@ -693,8 +668,7 @@ class model_base : public prob_grad {
    */
   virtual math::fvar<math::var> log_prob_jacobian(
       Eigen::Matrix<math::fvar<math::var>, -1, 1>& params_r,
-      std::ostream* msgs) const
-      = 0;
+      std::ostream* msgs) const = 0;
 
   /**
    * Return the log density for the specified unconstrained
@@ -707,8 +681,7 @@ class model_base : public prob_grad {
    */
   virtual math::fvar<math::var> log_prob_propto(
       Eigen::Matrix<math::fvar<math::var>, -1, 1>& params_r,
-      std::ostream* msgs) const
-      = 0;
+      std::ostream* msgs) const = 0;
 
   /**
    * Return the log density for the specified unconstrained
@@ -725,8 +698,7 @@ class model_base : public prob_grad {
    */
   virtual math::fvar<math::var> log_prob_propto_jacobian(
       Eigen::Matrix<math::fvar<math::var>, -1, 1>& params_r,
-      std::ostream* msgs) const
-      = 0;
+      std::ostream* msgs) const = 0;
 #endif
 };
 

--- a/src/stan/model/model_base.hpp
+++ b/src/stan/model/model_base.hpp
@@ -1,11 +1,11 @@
 #ifndef STAN_MODEL_MODEL_BASE_HPP
 #define STAN_MODEL_MODEL_BASE_HPP
 
-#include <stan/io/var_context.hpp>
-#include <stan/math/rev/core.hpp>
 #ifdef STAN_MODEL_FVAR_VAR
 #include <stan/math/mix.hpp>
 #endif
+#include <stan/io/var_context.hpp>
+#include <stan/math/rev/core.hpp>
 #include <stan/model/prob_grad.hpp>
 #include <stan/services/util/create_rng.hpp>
 #include <ostream>
@@ -78,7 +78,8 @@ class model_base : public prob_grad {
    */
   virtual void get_param_names(std::vector<std::string>& names,
                                bool include_tparams = true,
-                               bool include_gqs = true) const = 0;
+                               bool include_gqs = true) const
+      = 0;
   /**
    * WARNING: This function bakes in the assumption that the
    * parameter values are rectangular. This is already not true
@@ -107,7 +108,8 @@ class model_base : public prob_grad {
    */
   virtual void get_dims(std::vector<std::vector<size_t> >& dimss,
                         bool include_tparams = true,
-                        bool include_gqs = true) const = 0;
+                        bool include_gqs = true) const
+      = 0;
   /**
    *  Set the specified sequence to the indexed, scalar, constrained
    *  parameter names.  Each variable is output with a
@@ -148,7 +150,8 @@ class model_base : public prob_grad {
    */
   virtual void constrained_param_names(std::vector<std::string>& param_names,
                                        bool include_tparams = true,
-                                       bool include_gqs = true) const = 0;
+                                       bool include_gqs = true) const
+      = 0;
 
   /**
    * Set the specified sequence of parameter names to the
@@ -177,7 +180,8 @@ class model_base : public prob_grad {
    */
   virtual void unconstrained_param_names(std::vector<std::string>& param_names,
                                          bool include_tparams = true,
-                                         bool include_gqs = true) const = 0;
+                                         bool include_gqs = true) const
+      = 0;
 
   /**
    * Return the log density for the specified unconstrained
@@ -188,8 +192,8 @@ class model_base : public prob_grad {
    * @param[in,out] msgs message stream
    * @return log density for specified parameters
    */
-  virtual double log_prob(Eigen::VectorXd& params_r,
-                          std::ostream* msgs) const = 0;
+  virtual double log_prob(Eigen::VectorXd& params_r, std::ostream* msgs) const
+      = 0;
 
   /**
    * Return the log density for the specified unconstrained
@@ -201,7 +205,8 @@ class model_base : public prob_grad {
    * @return log density for specified parameters
    */
   virtual math::var log_prob(Eigen::Matrix<math::var, -1, 1>& params_r,
-                             std::ostream* msgs) const = 0;
+                             std::ostream* msgs) const
+      = 0;
 
   /**
    * Return the log density for the specified unconstrained
@@ -217,7 +222,8 @@ class model_base : public prob_grad {
    * @return log density for specified parameters
    */
   virtual double log_prob_jacobian(Eigen::VectorXd& params_r,
-                                   std::ostream* msgs) const = 0;
+                                   std::ostream* msgs) const
+      = 0;
 
   /**
    * Return the log density for the specified unconstrained
@@ -233,7 +239,8 @@ class model_base : public prob_grad {
    * @return log density for specified parameters
    */
   virtual math::var log_prob_jacobian(Eigen::Matrix<math::var, -1, 1>& params_r,
-                                      std::ostream* msgs) const = 0;
+                                      std::ostream* msgs) const
+      = 0;
 
   /**
    * Return the log density for the specified unconstrained
@@ -250,7 +257,8 @@ class model_base : public prob_grad {
    * @return log density for specified parameters
    */
   virtual double log_prob_propto(Eigen::VectorXd& params_r,
-                                 std::ostream* msgs) const = 0;
+                                 std::ostream* msgs) const
+      = 0;
 
   /**
    * Return the log density for the specified unconstrained
@@ -262,7 +270,8 @@ class model_base : public prob_grad {
    * @return log density for specified parameters
    */
   virtual math::var log_prob_propto(Eigen::Matrix<math::var, -1, 1>& params_r,
-                                    std::ostream* msgs) const = 0;
+                                    std::ostream* msgs) const
+      = 0;
 
   /**
    * Return the log density for the specified unconstrained
@@ -283,7 +292,8 @@ class model_base : public prob_grad {
    * @return log density for specified parameters
    */
   virtual double log_prob_propto_jacobian(Eigen::VectorXd& params_r,
-                                          std::ostream* msgs) const = 0;
+                                          std::ostream* msgs) const
+      = 0;
 
   /**
    * Return the log density for the specified unconstrained
@@ -299,7 +309,8 @@ class model_base : public prob_grad {
    * @return log density for specified parameters
    */
   virtual math::var log_prob_propto_jacobian(
-      Eigen::Matrix<math::var, -1, 1>& params_r, std::ostream* msgs) const = 0;
+      Eigen::Matrix<math::var, -1, 1>& params_r, std::ostream* msgs) const
+      = 0;
 
   /**
    * Convenience template function returning the log density for the
@@ -347,7 +358,8 @@ class model_base : public prob_grad {
    */
   virtual void transform_inits(const io::var_context& context,
                                Eigen::VectorXd& params_r,
-                               std::ostream* msgs) const = 0;
+                               std::ostream* msgs) const
+      = 0;
 
   /**
    * Convert the specified sequence of unconstrained parameters to a
@@ -370,7 +382,8 @@ class model_base : public prob_grad {
   virtual void write_array(stan::rng_t& base_rng, Eigen::VectorXd& params_r,
                            Eigen::VectorXd& params_constrained_r,
                            bool include_tparams = true, bool include_gqs = true,
-                           std::ostream* msgs = 0) const = 0;
+                           std::ostream* msgs = 0) const
+      = 0;
 
   /**
    * Convert the specified sequence of constrained parameters to a
@@ -385,7 +398,8 @@ class model_base : public prob_grad {
    */
   virtual void unconstrain_array(const Eigen::VectorXd& params_r_constrained,
                                  Eigen::VectorXd& params_r,
-                                 std::ostream* msgs = nullptr) const = 0;
+                                 std::ostream* msgs = nullptr) const
+      = 0;
 
   // TODO(carpenter): cut redundant std::vector versions from here ===
 
@@ -402,8 +416,8 @@ class model_base : public prob_grad {
    * @return log density for specified parameters
    */
   virtual double log_prob(std::vector<double>& params_r,
-                          std::vector<int>& params_i,
-                          std::ostream* msgs) const = 0;
+                          std::vector<int>& params_i, std::ostream* msgs) const
+      = 0;
 
   /**
    * Return the log density for the specified unconstrained
@@ -419,7 +433,8 @@ class model_base : public prob_grad {
    */
   virtual math::var log_prob(std::vector<math::var>& params_r,
                              std::vector<int>& params_i,
-                             std::ostream* msgs) const = 0;
+                             std::ostream* msgs) const
+      = 0;
 
   /**
    * Return the log density for the specified unconstrained
@@ -439,7 +454,8 @@ class model_base : public prob_grad {
    */
   virtual double log_prob_jacobian(std::vector<double>& params_r,
                                    std::vector<int>& params_i,
-                                   std::ostream* msgs) const = 0;
+                                   std::ostream* msgs) const
+      = 0;
 
   /**
    * Return the log density for the specified unconstrained
@@ -459,7 +475,8 @@ class model_base : public prob_grad {
    */
   virtual math::var log_prob_jacobian(std::vector<math::var>& params_r,
                                       std::vector<int>& params_i,
-                                      std::ostream* msgs) const = 0;
+                                      std::ostream* msgs) const
+      = 0;
 
   /**
    * Return the log density for the specified unconstrained
@@ -480,7 +497,8 @@ class model_base : public prob_grad {
    */
   virtual double log_prob_propto(std::vector<double>& params_r,
                                  std::vector<int>& params_i,
-                                 std::ostream* msgs) const = 0;
+                                 std::ostream* msgs) const
+      = 0;
 
   /**
    * Return the log density for the specified unconstrained
@@ -496,7 +514,8 @@ class model_base : public prob_grad {
    */
   virtual math::var log_prob_propto(std::vector<math::var>& params_r,
                                     std::vector<int>& params_i,
-                                    std::ostream* msgs) const = 0;
+                                    std::ostream* msgs) const
+      = 0;
 
   /**
    * Return the log density for the specified unconstrained
@@ -521,7 +540,8 @@ class model_base : public prob_grad {
    */
   virtual double log_prob_propto_jacobian(std::vector<double>& params_r,
                                           std::vector<int>& params_i,
-                                          std::ostream* msgs) const = 0;
+                                          std::ostream* msgs) const
+      = 0;
 
   /**
    * Return the log density for the specified unconstrained
@@ -541,7 +561,8 @@ class model_base : public prob_grad {
    */
   virtual math::var log_prob_propto_jacobian(std::vector<math::var>& params_r,
                                              std::vector<int>& params_i,
-                                             std::ostream* msgs) const = 0;
+                                             std::ostream* msgs) const
+      = 0;
 
   /**
    * Convenience template function returning the log density for the
@@ -596,7 +617,8 @@ class model_base : public prob_grad {
   virtual void transform_inits(const io::var_context& context,
                                std::vector<int>& params_i,
                                std::vector<double>& params_r,
-                               std::ostream* msgs) const = 0;
+                               std::ostream* msgs) const
+      = 0;
 
   /**
    * Convert the specified sequence of unconstrained parameters to a
@@ -621,7 +643,8 @@ class model_base : public prob_grad {
                            std::vector<int>& params_i,
                            std::vector<double>& params_r_constrained,
                            bool include_tparams = true, bool include_gqs = true,
-                           std::ostream* msgs = 0) const = 0;
+                           std::ostream* msgs = 0) const
+      = 0;
 
   /**
    * Convert the specified sequence of constrained parameters to a
@@ -636,7 +659,8 @@ class model_base : public prob_grad {
    */
   virtual void unconstrain_array(
       const std::vector<double>& params_r_constrained,
-      std::vector<double>& params_r, std::ostream* msgs = nullptr) const = 0;
+      std::vector<double>& params_r, std::ostream* msgs = nullptr) const
+      = 0;
 
 #ifdef STAN_MODEL_FVAR_VAR
 
@@ -651,7 +675,8 @@ class model_base : public prob_grad {
    */
   virtual math::fvar<math::var> log_prob(
       Eigen::Matrix<math::fvar<math::var>, -1, 1>& params_r,
-      std::ostream* msgs) const = 0;
+      std::ostream* msgs) const
+      = 0;
 
   /**
    * Return the log density for the specified unconstrained
@@ -668,7 +693,8 @@ class model_base : public prob_grad {
    */
   virtual math::fvar<math::var> log_prob_jacobian(
       Eigen::Matrix<math::fvar<math::var>, -1, 1>& params_r,
-      std::ostream* msgs) const = 0;
+      std::ostream* msgs) const
+      = 0;
 
   /**
    * Return the log density for the specified unconstrained
@@ -681,7 +707,8 @@ class model_base : public prob_grad {
    */
   virtual math::fvar<math::var> log_prob_propto(
       Eigen::Matrix<math::fvar<math::var>, -1, 1>& params_r,
-      std::ostream* msgs) const = 0;
+      std::ostream* msgs) const
+      = 0;
 
   /**
    * Return the log density for the specified unconstrained
@@ -698,7 +725,8 @@ class model_base : public prob_grad {
    */
   virtual math::fvar<math::var> log_prob_propto_jacobian(
       Eigen::Matrix<math::fvar<math::var>, -1, 1>& params_r,
-      std::ostream* msgs) const = 0;
+      std::ostream* msgs) const
+      = 0;
 #endif
 };
 

--- a/src/stan/model/model_base_crtp.hpp
+++ b/src/stan/model/model_base_crtp.hpp
@@ -1,10 +1,10 @@
 #ifndef STAN_MODEL_MODEL_BASE_CRTP_HPP
 #define STAN_MODEL_MODEL_BASE_CRTP_HPP
 
-#include <stan/model/model_base.hpp>
 #ifdef STAN_MODEL_FVAR_VAR
 #include <stan/math/mix.hpp>
 #endif
+#include <stan/model/model_base.hpp>
 #include <iostream>
 #include <utility>
 #include <vector>

--- a/src/stan/model/model_header.hpp
+++ b/src/stan/model/model_header.hpp
@@ -1,14 +1,14 @@
 #ifndef STAN_MODEL_MODEL_HEADER_HPP
 #define STAN_MODEL_MODEL_HEADER_HPP
 
+#include <stan/model/model_base.hpp>
+#include <stan/model/model_base_crtp.hpp>
 #include <stan/math.hpp>
 
 #include <stan/io/deserializer.hpp>
 #include <stan/io/serializer.hpp>
 
 #include <stan/model/rethrow_located.hpp>
-#include <stan/model/model_base.hpp>
-#include <stan/model/model_base_crtp.hpp>
 #include <stan/model/prob_grad.hpp>
 #include <stan/model/indexing.hpp>
 #include <stan/services/util/create_rng.hpp>

--- a/src/test/test-models/good/model/accumulate.stan
+++ b/src/test/test-models/good/model/accumulate.stan
@@ -1,0 +1,9 @@
+data {
+  int<lower=1> N;
+}
+parameters {
+  vector[N] y;
+}
+model {
+  target += y;
+}

--- a/src/test/unit/model/compile_test.cpp
+++ b/src/test/unit/model/compile_test.cpp
@@ -1,0 +1,2 @@
+// test model which previously failed with -DSTAN_MODEL_FVAR_VAR
+#include <test/test-models/good/model/accumulate.hpp>

